### PR TITLE
Spring Boot JPA task

### PR DESCRIPTION
**Pull request by Stelmaschyk**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/repository/BookRepository.java [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
+import org.springframework.data.jpa.repository.Query;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
+public interface BookRepository extends JpaRepository<Book, Long> {
+    @Query(value = "SELECT * FROM book WHERE is_deleted = false AND id = ?1", nativeQuery = true)
     Optional<Book> getBookById(Long id);
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/repository/BookRepository.java]: Is there a reason to use nativeQuery here? You can do it using default JPA method naming (something like findByIsDeletedIsFalseAndId

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -26,8 +27,8 @@ public BookDto save(CreateBookRequestDto requestDto) {
     @Override
     public List<BookDto> findAll() {
         return bookRepository.findAll().stream()
-                .map(bookMapper::toBookDto)
-                .toList();
+            .map(bookMapper::toBookDto)
+            .toList();
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]: Mapstruct supports mapping collections from the box

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -36,5 +37,24 @@ public BookDto findById(Long id) {
...
+    @Override
+    public BookDto updateBookById(Long id, UpdateBookRequestDto updateDto) {
+        Book existingBook = bookRepository.getBookById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id " + id));
+        existingBook.setTitle(updateDto.getTitle());
+        existingBook.setAuthor(updateDto.getAuthor());
+        existingBook.setIsbn(updateDto.getIsbn());
+        existingBook.setPrice(updateDto.getPrice());
+        existingBook.setDescription(updateDto.getDescription());
+        existingBook.setCoverImage(updateDto.getCoverImage());
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]: Can it be extracted to bookMapper.updateBook(existingBook, updateDto)

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 4]
```diff
@@ -8,9 +8,13 @@
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
 
 @Entity
 @Data
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]:
```suggestion
@Getter
@Setter
```

File: src/main/java/mate/academy/bookstore/repository/BookRepository.java [hunk 5]
```diff
@@ -1,13 +1,9 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> getBookById(Long id);
+public interface BookRepository extends JpaRepository<Book, Long> {
+    Optional<Book> findByIsDeletedIsFalseAndId(Long id);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 6]
```diff
@@ -25,16 +26,28 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
-                .map(bookMapper::toBookDto)
-                .toList();
+        List<Book> bookDtos = bookRepository.findAll();
+        return bookMapper.toBookDto(bookDtos);
     }
 
     @Override
     public BookDto findById(Long id) {
-        Book book = bookRepository.getBookById(id).orElseThrow(
+        Book book = bookRepository.findByIsDeletedIsFalseAndId(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
```

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 7]
```diff
@@ -25,16 +26,28 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
                 () -> new EntityNotFoundException("Can't find book by id " + id));
         return bookMapper.toBookDto(book);
     }
+
+    public BookDto updateBookById(Long id, UpdateBookRequestDto updateDto) {
+        Book existingBook = bookRepository.findByIsDeletedIsFalseAndId(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]:
```suggestion
        Book bookd = bookRepository.findById(id).orElseThrow(
```

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 8]
```diff
@@ -25,16 +26,28 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
+                () -> new EntityNotFoundException("Can't find book by id " + id));
+        Book updateBook = bookMapper.updateBook(updateDto);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]:
```suggestion
        bookMapper.updateBook(updateDto);
```

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 9]
```diff
@@ -25,16 +26,28 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
+        return bookMapper.toBookDto(bookRepository.save(updateBook));
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]:
```suggestion
        return bookMapper.toBookDto(bookRepository.save(book));
```

File: src/main/java/mate/academy/bookstore/mapper/BookMapper.java [hunk 10]
```diff
@@ -11,5 +13,9 @@
 public interface BookMapper {
     BookDto toBookDto(Book book);
 
+    List<BookDto> toBookDto(List<Book> book);
+
     Book toModel(CreateBookRequestDto requestDto);
+
+    Book updateBook(UpdateBookRequestDto updateBook);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/mapper/BookMapper.java]:
```suggestion
    void updateBookFromDto(UpdateBookRequestDto book, @MappingTarget Book entity);
```

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 11]
```diff
@@ -0,0 +1,42 @@
+databaseChangeLog:
+  - changeSet:
+      id: add-is_boolean-column-to-book-table
+      author: svg
+      changes:
+        - createTable:
+            tableName: book
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]:
```suggestion
            tableName: books
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 12]
```diff
@@ -8,9 +8,13 @@
...
+@SQLDelete(sql = "UPDATE book SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
 @Table(name = "book")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]:
```suggestion
@Table(name = "books")
```

File: src/main/resources/db/changelog/changes/02-add-is_boolean-column-to-book-table.yaml [hunk 13]
```diff
@@ -0,0 +1,14 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-books-table
+      author: svg
+      changes:
+        - addColumn:
+            tableName: book
+            columns:
+              - column:
+                  name: is_deleted
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-add-is_boolean-column-to-book-table.yaml]: replace to 01 changelog, addColumn isn't supported by autorollback

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**